src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final double GROW_VALUE = 1.5;
+    private int maxSize = 10;
+    private int actualSize = 0;
+    private Object[] array = new Object[maxSize];
+    private Object[] tempArray;
+
     @Override
     public void add(T value) {
-
+        if (actualSize == maxSize) {
+            grow();
+        }
+        array[actualSize] = value;
+        actualSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        if (actualSize == maxSize) {
+            grow();
+        }
+        tempArray = new Object[maxSize];
+        System.arraycopy(array, 0, tempArray, 0, index);
+        tempArray[index] = value;
+        System.arraycopy(array, index, tempArray, index + 1, actualSize - index);
+        actualSize++;
+        array = tempArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int requiredSize = maxSize - actualSize + list.size();
+        while (requiredSize > maxSize) {
+            grow();
+        }
+        tempArray = new Object[maxSize];
+        System.arraycopy(array, 0, tempArray, 0, actualSize);
+        System.arraycopy(toArray(list), 0, tempArray, actualSize, list.size());
+        actualSize = actualSize + list.size();
+        array = tempArray;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        final T removedItem = (T) array[index];
+        tempArray = new Object[maxSize];
+        actualSize--;
+        System.arraycopy(array, 0, tempArray, 0, index);
+        System.arraycopy(array, index + 1, tempArray, index, actualSize - index);
+        array = tempArray;
+        return removedItem;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < array.length; i++) {
+            if ((array[i] != null && array[i].equals(element))
+                    || (array[i] == null && element == null)) {
```

Comment:
remove extra parentheses
```suggestion
            if ((array[i] != null && array[i].equals(element))
                    || (array[i] == element)) {
```